xdsl/irdl/irdl.py

```diff
@@ -1805,51 +1926,61 @@ def get_operand_result_or_region(
         return args[begin_arg]
 
 
+def irdl_op_verify_regions(op: Operation, op_def: OpDef):
+    get_variadic_sizes(op, op_def, VarIRConstruct.REGION)
+    for i, (region, (name, region_def)) in enumerate(zip(op.regions, op_def.regions)):
+        if isinstance(region_def, SingleBlockRegionDef) and len(region.blocks) != 1:
+            raise VerifyException(
+                f"Region '{name}' at position {i} expected a single block, but got "
+                f"{len(region.blocks)} blocks"
+            )
+
+
 def irdl_op_verify_arg_list(
     op: Operation,
     op_def: OpDef,
-    construct: VarIRConstruct,
+    construct: Literal[VarIRConstruct.OPERAND, VarIRConstruct.RESULT],
     constraint_vars: dict[str, Attribute],
 ) -> None:
     """Verify the argument list of an operation."""
     arg_sizes = get_variadic_sizes(op, op_def, construct)
     arg_idx = 0
     var_idx = 0
-    args = get_op_constructs(op, construct)
+    args = cast(
```

Comment:
why the casts?

Reply:
get_op_constructs return `Sequence[SSAValue] | list[OpResult] | list[Region] | list[Successor]`, depending on the enum argument.
I could use overload for literal arguments, but not union of them, so used a cast.